agent/battle/runtime/runtime.py

```
@@ -0,0 +1,468 @@
+        mfaalog.info(f"[AutoBattle] _wait_until() scenes={[s.name for s in scenes]} timeout={timeout_s}s")
+        deadline = time.monotonic() + timeout_s
+        while time.monotonic() < deadline:
+            img = self.controller.post_screencap().wait().get()
+            scene = perception.detect_scene(self.ctx, img)
+            if scene in scenes:
+                mfaalog.info(f"[AutoBattle] _wait_until() matched scene={scene.name}")
+                return True
+            if tap_close:
+                self.executor.tap_top_right_close()
+            time.sleep(0.5)
+        mfaalog.info(f"[AutoBattle] _wait_until() timed out")
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)